### PR TITLE
Widget UI + configurator

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Widget UI improvements and adding of functionality
- Adds a configurator React app

Screenshot 2023-10-16 at 12 56 25

### CODE REVIEW

File: apps/cowswap-frontend/src/cow-react/index.tsx [hunk 1]
```diff
@@ -56,12 +58,18 @@ root.render(
                 <BlockNumberProvider>
                   <WithLDProvider>
                     <Updaters />
-                    <FeatureGuard featureFlag="cowFortuneEnabled">
-                      <FortuneWidget />
-                    </FeatureGuard>
+
+                    {!isInjectedWidgetMode && (
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/cow-react/index.tsx:62]: Don't render Fortune Widget and Appzi button for Widget mode

File: apps/cowswap-frontend/src/legacy/components/ErrorBoundary/index.tsx [hunk 2]
```diff
@@ -128,6 +130,8 @@ export default class ErrorBoundary extends React.Component<PropsWithChildren, Er
 
           return (
             <AppWrapper>
+
+            {!isInjectedWidget() &&
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/legacy/components/ErrorBoundary/index.tsx:134]: Hide the normal header (logo/menu/etc) in Widget mode

File: apps/cowswap-frontend/src/legacy/components/Header/index.tsx [hunk 3]
```diff
@@ -49,16 +50,31 @@ const CHAIN_CURRENCY_LABELS: { [chainId in ChainId]?: string } = {
   [ChainId.GNOSIS_CHAIN]: 'xDAI',
 }
 
+// Todo: fix 'any' types
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/legacy/components/Header/index.tsx:53]: Need help here fixing the types

File: apps/cowswap-frontend/src/modules/application/containers/TradeWidgetLinks/index.tsx [hunk 4]
```diff
@@ -36,41 +38,66 @@ const MENU_ITEMS: MenuItemConfig[] = [
...
+        {menuItem}
+      </FeatureGuard>
+    ) : (
+      menuItem
+    )
+  })
+
+  const singleMenuItem = menuItems.length === 1;
+
+  return isDropdown ? (
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/modules/application/containers/TradeWidgetLinks/index.tsx:86]: Render the menu in dropdown form if this is widget mode

File: apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/index.tsx [hunk 5]
```diff
@@ -179,14 +180,18 @@ export function OrdersTableContainer({
           <h3>
             <Trans>Connect a wallet</Trans>
           </h3>
-          <p>
-            <Trans>
-              To use {orderType} orders, please connect your wallet <br />
-              to one of our supported networks.
-            </Trans>
-          </p>
+          {!isInjectedWidget && (
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/index.tsx:183]: hide extra messaging in your orders overview when in widget mode

File: apps/widget-configurator/src/mui-theme-custom.d.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import '@mui/material/styles';
+
+declare module '@mui/material/styles' {
+  interface Palette {
+    cow: {
```
Comment (fairlighteth) [inline apps/widget-configurator/src/mui-theme-custom.d.ts:5]: adding custom palette items

File: apps/cowswap-frontend/src/common/hooks/useOrdersPanel.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { useState, useCallback } from 'react';
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/common/hooks/useOrdersPanel.ts:1]: creating a reusable hook to control the orders panel (user activity modal)

File: apps/cowswap-frontend/src/legacy/components/Header/index.tsx (hunk 3 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/components/Header/index.tsx]: Fixed

File: apps/cowswap-frontend/src/common/hooks/useOrdersPanel.ts (hunk 7 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/hooks/useOrdersPanel.ts]: Refactored it into `accountModalStateAtom`

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**